frontend/Nouns/CreateEditIntakeForm.tsx

```diff
@@ -0,0 +1,291 @@
+import { useReducer } from "react";
+import {
+  DragDropContext,
+  Droppable,
+  Draggable,
+  DropResult,
+} from "react-beautiful-dnd";
+import { RouterProps, useParams } from "react-router";
+import { Field } from "../../backend/DB/models/field";
+import { useQuery } from "react-query";
+import { Card } from "../Styleguide/Card";
+import { Loader } from "../Styleguide/Loader";
+import { Input } from "../Styleguide/Input";
+import { FormFieldLabel } from "../Styleguide/FormFieldLabel";
+import { FormField } from "../Styleguide/FormField";
+import { Modal } from "../Styleguide/Modal";
+
+export function CreateEditIntakeForm(props: RouterProps) {
+  const [state, dispatch] = useReducer<Reducer, State>(
+    reducer,
+    initialState,
+    () => initialState
+  );
+  const { nounId } = useParams<RouteParams>();
+  const { isLoading, isError, error } = useQuery<IntakeItem[]>(
+    `fields-${nounId}`,
+    async () => {
+      // const r = await flaxFetch<GetFieldsResponse>(`/api/nouns/${nounId}/fields`)
+      const r: GetIntakeItemsResponse = {
```

Comment:
This sample data will be replaced with an api request once the apis exist. Am working on the frontend first because I think it will help me sort through what the api should look like.

Reply:
This is a good approach.